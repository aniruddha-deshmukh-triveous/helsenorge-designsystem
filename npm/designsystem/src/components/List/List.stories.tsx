import React from 'react';

import { Title as DocsTitle, Subtitle, Description, Primary, ArgsTable, PRIMARY_STORY } from '@storybook/addon-docs';
import { Meta, StoryObj } from '@storybook/react';

import List from './List';
import { mediumLoremText } from '../../utils/loremtext';
import GridExample from '../GridExample';

const meta = {
  title: '@helsenorgeâˆ•designsystem-react/Components/List',
  component: List,
  parameters: {
    docs: {
      description: {
        component: 'Strukturert visning av punktvise data',
      },
      page: () => (
        <>
          <DocsTitle />
          <Subtitle />
          <Description />
          <Primary />
          <ArgsTable story={PRIMARY_STORY} />
        </>
      ),
    },
  },
  args: {
    children: '',
    variant: 'bullet',
    margin: false,
  },
  argTypes: {
    variant: {
      control: 'select',
      options: ['bullet', 'dashed', 'numbered', 'alphabetical'],
    },
    margin: {
      control: 'boolean',
    },
  },
} satisfies Meta<typeof List>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: args => (
    <GridExample>
      <List {...args}>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>
          {mediumLoremText}
          <List variant={args.variant}>
            <List.Item>{mediumLoremText}</List.Item>
            <List.Item>{mediumLoremText}</List.Item>
            <List.Item>{mediumLoremText}</List.Item>
            <List.Item>
              {mediumLoremText}
              <List variant={args.variant}>
                <List.Item>{mediumLoremText}</List.Item>
                <List.Item>{mediumLoremText}</List.Item>
                <List.Item>{mediumLoremText}</List.Item>
                <List.Item>
                  {mediumLoremText}
                  <List variant={args.variant}>
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>
                      {mediumLoremText}
                      <List variant={args.variant}>
                        <List.Item>{mediumLoremText}</List.Item>
                        <List.Item>{mediumLoremText}</List.Item>
                        <List.Item>{mediumLoremText}</List.Item>
                        <List.Item>{mediumLoremText}</List.Item>
                      </List>
                    </List.Item>
                  </List>
                </List.Item>
              </List>
            </List.Item>
          </List>
        </List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
        <List.Item>{mediumLoremText}</List.Item>
      </List>
    </GridExample>
  ),
};

export const Bullet: Story = {
  render: () => (
    <GridExample>
      <List variant="bullet">
        <List.Item>
          {mediumLoremText}
          <List variant="bullet">
            <List.Item>
              {mediumLoremText}
              <List variant="bullet">
                <List.Item>
                  {mediumLoremText}
                  <List variant="bullet">
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>{mediumLoremText}</List.Item>
                  </List>
                </List.Item>
                <List.Item>{mediumLoremText}</List.Item>
              </List>
            </List.Item>
            <List.Item>{mediumLoremText}</List.Item>
          </List>
        </List.Item>
        <List.Item>{mediumLoremText}</List.Item>
      </List>
    </GridExample>
  ),
};

export const Dashed: Story = {
  render: () => (
    <GridExample>
      <List variant="dashed">
        <List.Item>
          {mediumLoremText}
          <List variant="dashed">
            <List.Item>
              {mediumLoremText}
              <List variant="dashed">
                <List.Item>
                  {mediumLoremText}
                  <List variant="dashed">
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>{mediumLoremText}</List.Item>
                  </List>
                </List.Item>
                <List.Item>{mediumLoremText}</List.Item>
              </List>
            </List.Item>
            <List.Item>{mediumLoremText}</List.Item>
          </List>
        </List.Item>
        <List.Item>{mediumLoremText}</List.Item>
      </List>
    </GridExample>
  ),
};

export const Numbered: Story = {
  render: () => (
    <GridExample>
      <List variant="numbered">
        <List.Item>
          {mediumLoremText}
          <List variant="numbered">
            <List.Item>
              {mediumLoremText}
              <List variant="numbered">
                <List.Item>
                  {mediumLoremText}
                  <List variant="numbered">
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>{mediumLoremText}</List.Item>
                  </List>
                </List.Item>
                <List.Item>{mediumLoremText}</List.Item>
              </List>
            </List.Item>
            <List.Item>{mediumLoremText}</List.Item>
          </List>
        </List.Item>
        <List.Item>{mediumLoremText}</List.Item>
      </List>
    </GridExample>
  ),
};

export const Nested: Story = {
  render: () => (
    <GridExample>
      <List variant="numbered">
        <List.Item>
          {mediumLoremText}
          <List variant="bullet">
            <List.Item>
              {mediumLoremText}
              <List variant="alphabetical">
                <List.Item>
                  {mediumLoremText}
                  <List variant="dashed">
                    <List.Item>{mediumLoremText}</List.Item>
                    <List.Item>{mediumLoremText}</List.Item>
                  </List>
                </List.Item>
                <List.Item>{mediumLoremText}</List.Item>
              </List>
            </List.Item>
            <List.Item>{mediumLoremText}</List.Item>
          </List>
        </List.Item>
        <List.Item>{mediumLoremText}</List.Item>
      </List>
    </GridExample>
  ),
};
