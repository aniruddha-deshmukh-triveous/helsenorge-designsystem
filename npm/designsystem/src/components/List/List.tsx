import React from 'react';

import classNames from 'classnames';

import listStyles from './styles.module.scss';
import { AnalyticsId } from '../../constants';

interface ListProps {
  children: React.ReactNode[];
  className?: string;
  /** Sets the data-testid attribute. */
  testId?: string;
}

function List(props: ListProps) {
  const { children, className, testId } = props;
  return (
    <ul className={classNames(listStyles.list, className)} data-testid={testId} data-analyticsid={AnalyticsId.List}>
      {children.map((child: React.ReactNode, index: number) => {
        return (
          <li className={listStyles.list__item} key={index}>
            {child}
          </li>
        );
      })}
    </ul>
  );
}

export default List;
