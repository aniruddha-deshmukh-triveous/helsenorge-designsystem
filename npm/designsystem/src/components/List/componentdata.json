{
  "props": {
    "className": {
      "defaultValue": null,
      "description": "",
      "name": "className",
      "parent": {
        "fileName": "src/components/List/List.tsx",
        "name": "ListProps"
      },
      "declarations": [
        {
          "fileName": "src/components/List/List.tsx",
          "name": "ListProps"
        }
      ],
      "required": false,
      "type": {
        "name": "string"
      }
    }
  }
}
