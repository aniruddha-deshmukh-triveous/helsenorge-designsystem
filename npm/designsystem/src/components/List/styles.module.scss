@import '../../scss/_palette.scss';

.list {
  width: 100%;
  list-style: none;
  padding: 0 1rem 0 0.5rem;

  &__item {
    border-bottom: 1px solid $neutral600;
  }
}
