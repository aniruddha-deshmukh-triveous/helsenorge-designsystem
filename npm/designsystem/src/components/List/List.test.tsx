import React from 'react';

import { render, screen } from '@testing-library/react';

import List from './List';

test('displays a list of 4 elements', (): void => {
  const { container, getAllByRole } = render(
    <List>
      <h1>Tjeneste 1</h1>
      <h1>Tjeneste 2</h1>
      <h1>Tjeneste 3</h1>
      <h1>Tjeneste 4</h1>
    </List>
  );
  expect(getAllByRole('heading').length).toEqual(4);
  expect(container).toMatchSnapshot();
});

describe('Gitt at List skal vises', (): void => {
  describe('Når testId-prop er satt', (): void => {
    test('Så kan komponenten finnes ved hjelp av testId', (): void => {
      render(
        <List testId="bare-tester">
          <h1>Tjeneste 1</h1>
          <h1>Tjeneste 2</h1>
          <h1>Tjeneste 3</h1>
          <h1>Tjeneste 4</h1>
        </List>
      );

      const component = screen.getByTestId('bare-tester');
      expect(component).toBeVisible();
    });
  });
});
