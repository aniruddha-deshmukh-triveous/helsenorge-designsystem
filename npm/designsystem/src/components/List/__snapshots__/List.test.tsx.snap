// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a list of 4 elements 1`] = `
<div>
  <ul
    class="list"
    data-analyticsid="list"
  >
    <li
      class="list__item"
    >
      <h1>
        Tjeneste 1
      </h1>
    </li>
    <li
      class="list__item"
    >
      <h1>
        Tjeneste 2
      </h1>
    </li>
    <li
      class="list__item"
    >
      <h1>
        Tjeneste 3
      </h1>
    </li>
    <li
      class="list__item"
    >
      <h1>
        Tjeneste 4
      </h1>
    </li>
  </ul>
</div>
`;
